src/status_im/contexts/profile/edit/name/view.cljs

```diff
@@ -0,0 +1,66 @@
+(ns status-im.contexts.profile.edit.name.view
+  (:require [clojure.string :as string]
+            [quo.core :as quo]
+            [react-native.core :as rn]
+            [react-native.safe-area :as safe-area]
+            [reagent.core :as reagent]
+            [status-im.constants :as constants]
+            [status-im.contexts.profile.edit.name.style :as style]
+            [status-im.contexts.profile.edit.name.utils :as utils]
+            [status-im.contexts.profile.utils :as profile.utils]
+            [utils.debounce :as debounce]
+            [utils.i18n :as i18n]
+            [utils.re-frame :as rf]))
+
+(defn view
+  []
+  (let [insets              (safe-area/get-insets)
+        profile             (rf/sub [:profile/profile-with-image])
+        customization-color (rf/sub [:profile/customization-color])
+        display-name        (profile.utils/displayed-name profile)
+        full-name           (reagent/atom display-name)
+        error-msg           (reagent/atom nil)
+        validate-name       (debounce/debounce #(reset! error-msg
+                                                  (utils/validation-name %))
+                                               500)
+        on-change-text      (fn [s]
+                              (reset! full-name (string/trim s))
+                              (validate-name s))]
+    (fn []
+      [quo/overlay
+       {:type            :shell
+        :container-style (style/page-wrapper insets)}
+       [quo/page-nav
+        {:key        :header
+         :background :blur
+         :icon-name  :i/arrow-left
+         :on-press   #(rf/dispatch [:navigate-back])}]
+       [rn/keyboard-avoiding-view
+        {:key   :content
+         :style style/screen-container}
+        [rn/view {:style {:gap 22}}
+         [quo/text-combinations {:title (i18n/label :t/name)}]
+         [quo/input
+          {:theme           :dark
+           :blur?           true
+           :error?          (seq @error-msg)
+           :container-style {:margin-bottom -11}
+           :default-value   @full-name
+           :auto-focus      true
+           :char-limit      constants/profile-name-max-length
+           :label           (i18n/label :t/profile-name)
+           :on-change-text  on-change-text}]
+         (when (seq @error-msg)
```

Comment:
I know strings support `seq`, but `seq` on a string is more expensive if all you want is to check if it's not blank.

The cheaper way is `(when-not (string/blank? @error-msg))`. Benchamarks on the JVM, so take with a grain of salt:

```clojure
(let [s "hello"]
  (criterium/with-progress-reporting (criterium/bench (seq s) :verbose)))
;; =>
;; Evaluation count : 1350768960 in 60 samples of 22512816 calls.
;; Execution time sample mean : 40.908934 ns
;; Execution time mean : 40.907411 ns
;; Execution time sample std-deviation : 0.752860 ns
;; Execution time std-deviation : 0.760289 ns
;; Execution time lower quantile : 39.410556 ns ( 2.5%)
;; Execution time upper quantile : 42.067256 ns (97.5%)
;; Overhead used : 5.337987 ns

(let [s "hello"]
  (criterium/with-progress-reporting (criterium/bench (not (string/blank? s)) :verbose)))
;; =>
;; Evaluation count : 8763828660 in 60 samples of 146063811 calls.
;; Execution time sample mean : 2.086615 ns
;; Execution time mean : 2.086406 ns
;; Execution time sample std-deviation : 0.041965 ns
;; Execution time std-deviation : 0.043085 ns
;; Execution time lower quantile : 1.952785 ns ( 2.5%)
;; Execution time upper quantile : 2.125713 ns (97.5%)
;; Overhead used : 5.337987 ns
```